src/App.tsx

```diff
@@ -1,111 +1,321 @@
+      />
+      <div className="col-span-2">
+        <CategoricalField
+          label="Sex"
+          options={Object.entries(sexes).map(([key, value]) => ({
+            label: key,
+            id: value,
+          }))}
+          onFieldChange={(label, value) => updateQueryParams(label, value)}
+        />
+      </div>
+      <div className="col-span-2 row-start-3">
+        <div className="grid grid-cols-12 items-center gap-4">
+          <div className="col-span-9">
+            <CategoricalField
+              label="Diagnosis"
+              options={diagnosisOptions.map((d) => ({
+                label: d.Label,
+                id: d.TermURL,
+              }))}
+              onFieldChange={(label, value) => updateQueryParams(label, value)}
+            />
+          </div>
+          <div>
+            <FormControlLabel
+              control={<Checkbox name="healthyControl" />}
+              label="Healthy Control"
+              onChange={() => setIsControl(!isControl)}
+            />
+          </div>
+        </div>
+      </div>
+      <div className="col-span-2 row-start-4">
+        <ContinuousField
+          label="Minimum number of sessions"
+          onFieldChange={(label, value) => updateQueryParams(label, value)}
+        />
+      </div>
+      <div className="col-span-2 row-start-5">
+        <CategoricalField
+          label="Assessment tool"
+          options={assessmentOptions.map((a) => ({ label: a.Label, id: a.TermURL }))}
+          onFieldChange={(label, value) => updateQueryParams(label, value)}
+        />
+      </div>
+      <div className="col-span-2 row-start-6">
+        <CategoricalField
+          label="Imaging modality"
+          options={Object.entries(modalityOptions).map(([, value]) => ({
+            label: value.label,
+            id: value.TermURL,
+          }))}
+          onFieldChange={(label, value) => updateQueryParams(label, value)}
+        />
+      </div>
+      <div className="row-start-7">
+        <Button
+          variant="contained"
+          endIcon={<SendIcon />}
+          onClick={() => onSubmitQuery(constructQueryURL())}
+        >
+          Submit Query
+        </Button>
+      </div>
+    </div>
+  );
+}
+
+function App() {
+
+  const [diagnosisOptions, setDiagnosisOptions] = useState<APIOption[]>([]);
+  const [assessmentOptions, setAssessmentOptions] = useState<APIOption[]>([]);
+  const [result, setResult] = useState<Result[]>([]);
+
+  useEffect( () => {
+    async function fetchData(dataElementURI : string, setState : (options: APIOption[]) => void) {
+      try {
+        const response: AxiosResponse<RetrievedOption> = await axios.get(`${attributesURL}${dataElementURI}`);
+        if (response.data[dataElementURI].length === 0) {
+          // TODO: make into a toast
+        } else {
+          setState(response.data[dataElementURI].sort((a, b) => a.Label.localeCompare(b.Label)));
```

Comment:
I think this would be a lot easier to read over if instead of setting state directly from inside the function we just returned the response and then handle the `setState` business outside - e.g. we wouldn't have these side-effect calls to just `fetchData(something)` that don't make clear what will happen after the fetch. We should also move the function out of the useEffect hook to keep it tight.

Reply:
> I think this would be a lot easier to read over if instead of setting state directly from inside the function we just returned the response and then handle the setState business outside

Since its an `async` function it returns a promise which requires another `async-wait` to set the state since the `seState` functions don't allow for promises.

> We should also move the function out of the useEffect hook to keep it tight.

Tried that as we discussed but Eslint didn't seem to like it. Since the `fetch` function wasn't defined inside of the hook it asked to either add it to the dependency array which would no longer be the equivalent of `componentDidMount` or remove it. A workaround suggested by the GPT was to use [`useCallBack`](https://react.dev/reference/react/useCallback) and put it in the dependency array of that and then use it in the `useEffect` which upon searching a bit seemed like an overkill and didn't seems appropriate use of `useCallback`.